modules/templates/templates_api.rb

```diff
@@ -14,8 +14,8 @@ class Proxy::TemplatesApi < Sinatra::Base
   end
 
   get "/:kind" do |kind|
-    log_halt(500, "Failed to retrieve #{kind} template for #{params[:token]}: ") do
-      Proxy::Templates::Handler.get_template(kind, params[:token], params[:static])
+    log_halt(500, "Failed to retrieve #{kind} template for #{params.inspect}: ") do
+      Proxy::Templates::Handler.get_template(kind, params, headers)
```

Comment:
`Sinatra::Helpers#headers` is for setting response headers, and returns current response headers when called without parameters: http://www.rubydoc.info/gems/sinatra/Sinatra/Helpers#headers-instance_method. Rack stores request headers in `request.env` hash, under keys starting with `HTTP_` (http://rack.rubyforge.org/doc/SPEC.html, lookup "HTTP_ Variables").